### PR TITLE
ceph: add support for tls certs via k8s tls secrets for rgw

### LINKED ISSUE #2079: allow ObjectStore -> sslCertificateRef with standard kubernetes.io/tls certificates
Allow ObjectStore -> sslCertificateRef to use cert-manager genereated certificates.
Right now the certificate looks like:
```
{
  tls.crt: ...
  tls.key: ...
}
```
according to the docs it only eats combined pem certificates.

It would be awesome it would directly convert or eat those standardly generated acme cert-manager certificates. Because we have loads of them freely.

### PR DESCRIPTION
**Pull request by thotz**
**Description of your changes:**
With this PR the RGW can accept TLS certs as K8s TLS secrets

Signed-off-by: Jiffin Tony Thottan <thottanjiffin@gmail.com>

**Which issue is resolved by this Pull Request:**
Resolves #2079

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

File: pkg/operator/ceph/object/spec.go [hunk 1]
```diff
@@ -492,29 +493,48 @@ func getLabels(name, namespace string, includeNewLabels bool) map[string]string
...
 	userReadOnly := int32(0444)
 	var secretVolSrc *v1.SecretVolumeSource
-	if objSpec.Gateway.SSLCertificateRef != "" {
+	if c.store.Spec.Gateway.SSLCertificateRef != "" {
 		secretVolSrc = &v1.SecretVolumeSource{
-			SecretName: objSpec.Gateway.SSLCertificateRef,
-			Items: []v1.KeyToPath{
+			SecretName: c.store.Spec.Gateway.SSLCertificateRef,
+		}
+		if c.rgwTLSSecretType() == "" {
```
Comment (leseb) [inline pkg/operator/ceph/object/spec.go:507]: use switch case?

File: pkg/operator/ceph/object/spec.go [hunk 2]
```diff
@@ -492,29 +493,48 @@ func getLabels(name, namespace string, includeNewLabels bool) map[string]string
...
 			}}
 	} else {
 		return nil, errors.New("no TLS certificates found")
 	}
 
 	return secretVolSrc, nil
 }
+
+func (c *clusterConfig) rgwTLSSecretType() v1.SecretType {
+	ctx := context.TODO()
```
Comment (leseb) [inline pkg/operator/ceph/object/spec.go:534]: you can directly cast `context.TODO()` below, no need for a new variable.

File: pkg/operator/ceph/object/spec.go [hunk 3]
```diff
@@ -492,29 +493,48 @@ func getLabels(name, namespace string, includeNewLabels bool) map[string]string
...
+	rgwTlsSecret, _ := c.context.Clientset.CoreV1().Secrets(c.clusterInfo.Namespace).Get(ctx, c.store.Spec.Gateway.SSLCertificateRef, metav1.GetOptions{})
+	if rgwTlsSecret != nil {
```
Comment (leseb) [inline pkg/operator/ceph/object/spec.go:536]: Why not return an error since the caller can handle errors?

**Review by leseb — CHANGES_REQUESTED** [review]

**Review by leseb — CHANGES_REQUESTED** [review]
Please add a release note

**Review by leseb — APPROVED** [review]

**Comment by oaudry** [comment]
great job ! many thx

**Pull request merged**